packages/relay/src/lib/eth.ts

```diff
@@ -1630,6 +1613,61 @@ export class EthImpl implements Eth {
           .then();
       }
     }
+
+    // After the try-catch process above, the `formattedTransactionId` is potentially valid in only two scenarios:
+    //   - The transaction was successfully submitted and fully processed by CN and MN.
+    //   - The transaction encountered "SDK timeout exceeded" or "Connection Dropped" errors from the SDK but still potentially reached the consensus level.
+    // In both scenarios, polling the MN is required to verify the transaction's validity before return the transaction hash to clients.
+    if (formattedTransactionId) {
+      try {
+        const contractResult = await this.mirrorNodeClient.repeatedRequest(
+          this.mirrorNodeClient.getContractResult.name,
+          [formattedTransactionId, requestDetails],
+          this.mirrorNodeClient.getMirrorNodeRequestRetryCount(),
+          requestDetails,
+        );
+
+        if (!contractResult) {
+          if (
+            sendRawTransactionError instanceof SDKClientError &&
+            (sendRawTransactionError.isConnectionDropped() || sendRawTransactionError.isTimeoutExceeded())
+          ) {
+            throw sendRawTransactionError;
+          }
+
+          this.logger.warn(
+            `${requestIdPrefix} No matching transaction record retrieved: transactionId=${submittedTransactionId}`,
+          );
+
+          throw predefined.INTERNAL_ERROR(
+            `No matching transaction record retrieved: transactionId=${submittedTransactionId}`,
+          );
+        }
+
+        if (contractResult.hash == null) {
+          this.logger.error(
+            `${requestIdPrefix} Transaction returned a null transaction hash: transactionId=${submittedTransactionId}`,
+          );
+          throw predefined.INTERNAL_ERROR(
+            `Transaction returned a null transaction hash: transactionId=${submittedTransactionId}`,
+          );
```

Comment:
missing test case

Reply:
Ah good point! Well it was from the previous implementation so I thought it was covered! Will add tests where appropriate!